addons/stock/models/stock_orderpoint.py

```diff
@@ -320,27 +320,39 @@ def _search_qty_to_order(self, operator, value):
 
     @api.depends('qty_multiple', 'qty_forecast', 'product_min_qty', 'product_max_qty', 'visibility_days')
     def _compute_qty_to_order_computed(self):
-        for orderpoint in self:
-            if not orderpoint.product_id or not orderpoint.location_id:
-                orderpoint.qty_to_order_computed = False
-                continue
-            qty_to_order = 0.0
+        def to_compute(orderpoint):
             rounding = orderpoint.product_uom.rounding
             # The check is on purpose. We only want to consider the visibility days if the forecast is negative and
             # there is a already something to ressuply base on lead times.
-            if float_compare(orderpoint.qty_forecast, orderpoint.product_min_qty, precision_rounding=rounding) < 0:
-                # We want to know how much we should order to also satisfy the needs that gonna appear in the next (visibility) days
-                product_context = orderpoint._get_product_context(visibility_days=orderpoint.visibility_days)
-                qty_forecast_with_visibility = orderpoint.product_id.with_context(product_context).read(['virtual_available'])[0]['virtual_available'] + orderpoint._quantity_in_progress()[orderpoint.id]
-                qty_to_order = max(orderpoint.product_min_qty, orderpoint.product_max_qty) - qty_forecast_with_visibility
-                remainder = orderpoint.qty_multiple > 0.0 and qty_to_order % orderpoint.qty_multiple or 0.0
-                if (float_compare(remainder, 0.0, precision_rounding=rounding) > 0
-                        and float_compare(orderpoint.qty_multiple - remainder, 0.0, precision_rounding=rounding) > 0):
-                    if float_is_zero(orderpoint.product_max_qty, precision_rounding=rounding):
-                        qty_to_order += orderpoint.qty_multiple - remainder
-                    else:
-                        qty_to_order -= remainder
+            return (
+                orderpoint.product_id
+                and orderpoint.location_id
+                and float_compare(orderpoint.qty_forecast, orderpoint.product_min_qty, precision_rounding=rounding) < 0
+            )
+
+        orderpoints = self.filtered(to_compute)
+        orderpoints_by_location = {}
+        for location, grouped_orderpoints in orderpoints.grouped(lambda o: o.location_id).items():
+            orderpoints_by_location[location] = grouped_orderpoints
+        qty_in_progress_by_orderpoint = {}
+        for orderpoint_group in orderpoints_by_location.values():
+            qty_in_progress_by_orderpoint.update(orderpoint_group._quantity_in_progress())
```

Comment:
```suggestion
        qty_in_progress_by_orderpoint = orderpoints._quantity_in_progress()
```
Did I miss where it require the orderpoints to be grouped by locations at this point?

Reply:
Yeah that's the whole point of this patch. The test `test_multi_lingual_orderpoints` introduced here: https://github.com/odoo/odoo/commit/606dc7187a5cfefa66316e0850919766d466c226 is currently failing if we don't batch quantity_in_progress by location_id. The issue seems to be located somewhere in `purchase_stock/product.py:_get_quantity_in_progress` although I'm not really sure where. It seems that this method is not properly batched but at the same time the new test is the only one failing when batching all the orderpoints 😟